docs/best-practices/java-coding-guidelines/readme.md

```diff
@@ -595,6 +594,20 @@ ImmutableList.Builder<String> l = new ImmutableList.Builder<>();
 ImmutableList.Builder<String> l = ImmutableList.builder();
 ```
 
+Guava collection factory methods such as `Maps.newHashMap`,
+`Lists.newArrayList` et al, should however be considered deprecated
+and should not be used.
+
+``` java
+// BAD. Don't do this.
+List<String> list = Lists.newArrayList();
```

Comment:
to give this some teeth, @aioobe , should we add an errorprone check to WARN or ERROR against this?

Reply:
I would be in favor of it. @schlosna has already outlined an implementation? Should I include the proposed code in this PR?